IdentityCore/src/validation/MSIDAuthority.h

```diff
@@ -17,57 +15,75 @@
 //
 // THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 // IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 // AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
-//
-//------------------------------------------------------------------------------
 
 #import <Foundation/Foundation.h>
 #import "MSIDAuthorityResolving.h"
+#import "MSIDCache.h"
+
+extern NSString * _Nonnull const MSIDTrustedAuthority;
+extern NSString * _Nonnull const MSIDTrustedAuthorityWorldWide;
 
 @class MSIDOpenIdProviderMetadata;
 
 typedef void(^MSIDOpenIdConfigurationInfoBlock)(MSIDOpenIdProviderMetadata * _Nullable metadata, NSError * _Nullable error);
 
-extern NSString * _Nonnull const MSIDTrustedAuthorityWorldWide;
+@interface MSIDAuthority : NSObject <NSCopying>
+{
+@protected
+    NSURL *_url;
+    NSURL *_openIdConfigurationEndpoint;
+}
+
+@property (class, readonly, nonnull) MSIDCache *openIdConfigurationCache;
+
+@property (nonatomic, readonly, nonnull) NSURL *url;
+
+@property (nonatomic, readonly, nullable) NSURL *openIdConfigurationEndpoint;
+
+@property (nonatomic, readonly, nullable) MSIDOpenIdProviderMetadata *metadata;
+
+- (instancetype _Nullable )init NS_UNAVAILABLE;
++ (instancetype _Nullable )new NS_UNAVAILABLE;
+
+- (nullable instancetype)initWithURL:(nonnull NSURL *)url
+                             context:(nullable id<MSIDRequestContext>)context
+                               error:(NSError * _Nullable __autoreleasing * _Nullable)error NS_DESIGNATED_INITIALIZER;
+
+- (nullable instancetype)initWithURL:(nonnull NSURL *)url
+                           rawTenant:(nullable NSString *)rawTenant
```

Comment:
Do we need initializer with tenant in the base class?

Reply:
Removed it from MSIDAuthority.